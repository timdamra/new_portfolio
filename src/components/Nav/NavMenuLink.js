import React from 'react'
import { useHistory } from 'react-router-dom'
import PropTypes from 'prop-types'

export function NavMenuLink(props) {
  let history = useHistory()

  return (
    <div
      className={props.isMobile ? 'nav__extended--link' : 'nav--link'}
      onClick={() => {
        props.setNavIsExtended(false)
        history.push(props.link)
      }}
    >
      {props.label}
    </div>
  )
}

NavMenuLink.propTypes = {
  label: PropTypes.string,
  link: PropTypes.string,
  setNavIsExtended: PropTypes.func,
  isMobile: PropTypes.bool
}

export default NavMenuLink
