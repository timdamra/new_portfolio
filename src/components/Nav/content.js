export const LABELS = [
    { label: 'Home', link: '/' },
    { label: 'Portfolio', link: '/portfolio' },
    { label: 'Contact', link: '/contact' }
]
