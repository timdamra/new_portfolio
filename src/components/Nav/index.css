.nav {
    background-color: silver;
    display: flex;
    flex-direction: row;
}

.nav--icon {
    font-size: 2em;
}

.nav__extended {       
    flex-direction: column;
    align-items: center;
}

.nav--link {
    padding: 4px 8px;
}

.nav__extended--link {
    text-align: center;
    border-bottom: 1px solid black;
    background-color: darkgray;
}

@media screen and (min-width: 768px) {
    .nav {
        align-items: stretch;
        justify-content: space-evenly;
        width: 100%;
    }
}
