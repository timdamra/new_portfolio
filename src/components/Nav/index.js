import React, { useState } from 'react'

import NavMenuLink from './NavMenuLink'
import { LABELS } from './content'
import './index.css'

export function Nav() {
    const isMobile = window.innerWidth < 768
    const cssStyles = isMobile ? 'nav__extended' : 'nav'      
    const [navIsExtended, setNavIsExtended] = useState(false) 

    const extendedMobile = (isMobile && navIsExtended)
    const displayNav = extendedMobile || (!isMobile && !navIsExtended)    

    const shouldDisplayNavIcons = () => {
        if (isMobile) {
            return navIsExtended ? (
                <div className="nav--icon m-b-8" onClick={() => setNavIsExtended(false)}>&times;</div>
            ) : (
                <div className="nav--icon m-b-8" onClick={() => setNavIsExtended(true)}>&#9776;</div>
            )
        }
    }

    return (
        <nav className={`${cssStyles} p-t-8`}>
            {shouldDisplayNavIcons()}
            {
                displayNav &&
                LABELS.length > 0 && 
                LABELS.map(navItem => (
                    <NavMenuLink 
                      label={navItem.label}
                      key={navItem.label}
                      link={navItem.link}
                      setNavIsExtended={setNavIsExtended}
                      isMobile={isMobile}
                    />
                ))
            }
        </nav>
    )
}

export default Nav
