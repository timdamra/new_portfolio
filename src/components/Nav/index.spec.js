import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

import Nav from './index'

const className = 'nav p-t-8'

describe('Nav', () => {
  let wrapper

  beforeAll(() => {
    wrapper = mount(
      <MemoryRouter>
        <Nav />
      </MemoryRouter>
    )
  })

  afterAll(() => {
    wrapper.unmount()
  })

  describe('Component', () => {
    it('should render nav with correct className', () => {
      expect(wrapper.find('nav').props().className).toBe(className)
    })

    it('should render nav with 3 links', () => {
      const navLinks = wrapper.find('.nav--link')
      expect(navLinks.length).toBe(3)
    })

    it('should navigate to new page upon clicking nav item', () => {
      const homeLink = wrapper.find('.nav--link').at(1)

      expect(wrapper.find('Router').props().history.entries.length).toBe(1)
      expect(wrapper.find('Router').props().history.entries[0].pathname).toBe(
        '/'
      )

      homeLink.simulate('click')

      expect(wrapper.find('Router').props().history.entries.length).toBe(2)
      expect(wrapper.find('Router').props().history.entries[0].pathname).toBe(
        '/'
      )
      expect(wrapper.find('Router').props().history.entries[1].pathname).toBe(
        '/portfolio'
      )
    })
  })

  describe('Snapshot', () => {
    it('should match snapshot', () => {
      const component = renderer.create(wrapper)
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
