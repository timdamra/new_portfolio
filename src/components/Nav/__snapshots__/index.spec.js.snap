// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Snapshot should match snapshot 1`] = `
<nav
  className="nav p-t-8"
>
  <div
    className="nav--link"
    onClick={[Function]}
  >
    HOME
  </div>
  <div
    className="nav--link"
    onClick={[Function]}
  >
    MY WORK
  </div>
  <div
    className="nav--link"
    onClick={[Function]}
  >
    CONTACT
  </div>
</nav>
`;
